src/Infrastructure/Repositories/LocalFilesRepository.php

```diff
@@ -53,13 +53,7 @@ public function getFiles(): iterable
      */
     public function within(string $directory, array $exclude): FilesRepository
     {
-        $this->finder->in([$directory])->exclude($exclude);
-
-        foreach ($exclude as $value) {
-            if (substr($value, -4) === '.php') {
-                $this->finder->notName($value);
-            }
-        }
+        $this->finder->in([$directory])->notPath($exclude);
```

Comment:
There is no breaking changes on this?

Reply:
In fact, that might be BC if some people use regexp for exclusion...

Actually, we can exclude folder (thanks to ->exclude), or exclude files that corresponding a pattern (with ->notName), but we cannot exclude a file by providing it's path

```php
#phpinsights.php 
    'exclude' => [
        'Kernel.php', // => ✅, will exclude ALL Kernel.php found in every directory
        'stubs', // => ✅, will exclude stubs 
        'src/Domain/Kernel.php', // => ❌ the file is not excluded
    ],
```

By replacing ->exclude by ->notPath we can cover all the case, and keep using ->notName for use regexp. 

@Sixine, sorry for giving you a false direction, can you re-add the foreach (and maybe provide some tests with regex patterns ? :innocent:)

Reply:
Sorry for the delay, it is done! :)

Reply:
@Jibbarth can you validate this pull request buddy? Thanks! Make sure that there is no breaking changes here.